lib/wallet/balance-tracker-contract.spec.js

```diff
@@ -28,42 +28,45 @@ const fakeEthers = {
     Contract
 };
 
-const BalanceTrackerContract = proxyquire('./balance-tracker-contract', {
-    'ethers': fakeEthers,
-    './abis/ropsten/BalanceTracker': fakeBalanceTrackerDeployment
-});
-
-const fakeProvider = {
-    network: {
-        chainId: '123456789',
-        name: 'ropsten'
-    }
-};
-const fakeWallet = {
-    provider: fakeProvider
-};
-
 describe('BalanceTrackerContract', () => {
-
-    [
-        ['wallet', fakeWallet],
-        ['provider', fakeProvider]
-    ].forEach(([description, walletOrProvider]) => {
-        context('with ' + description, () => {
-            let contract;
-
-            beforeEach(() => {
-                contract = new BalanceTrackerContract(walletOrProvider);
+    ['ropsten', 'homestead'].forEach((network) => {
+        describe(network, () => {
+            const abstractionModule = network === 'ropsten' ? 'nahmii-contract-abstractions-ropsten' : 'nahmii-contract-abstractions';
+            const BalanceTrackerContract = proxyquire('./balance-tracker-contract', {
+                'ethers': fakeEthers,
+                [`${abstractionModule}/build/contracts/BalanceTracker.json`]: fakeBalanceTrackerDeployment
             });
-
-            it('is an instance of Contract', () => {
-                expect(contract).to.be.an.instanceOf(Contract);
-            });
-
-            it('uses correct contract address in construction', () => {
-                expect(contract.addr).to.eql(fakeBalanceTrackerDeployment.networks['123456789'].address);
-                expect(contract.abi).to.eql(fakeBalanceTrackerDeployment.abi);
-                expect(contract.signerOrProvider).to.equal(walletOrProvider);
+    
+            const fakeProvider = {
+                network: {
+                    chainId: '123456789',
+                    name: network
+                }
+            };
+            const fakeWallet = {
+                provider: fakeProvider
+            };
+            [
+                ['wallet', fakeWallet],
+                ['provider', fakeProvider]
+            ].forEach(([description, walletOrProvider]) => {
+                context('with ' + description, () => {
+                    let contract;
+        
+                    beforeEach(() => {
+                        contract = new BalanceTrackerContract(walletOrProvider);
+                    });
+        
+                    it('is an instance of Contract', () => {
+                        expect(contract).to.be.an.instanceOf(Contract);
+                    });
+        
+                    it('uses correct contract address in construction', () => {
+                        expect(contract.addr).to.eql(fakeBalanceTrackerDeployment.networks['123456789'].address);
+                        expect(contract.abi).to.eql(fakeBalanceTrackerDeployment.abi);
+                        expect(contract.signerOrProvider).to.equal(walletOrProvider);
+                    });
+                });
             });
         });
     });
```

Comment:
Maybe also add the test for an "unknown" network, like you have in the settlement sub-module?